src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/CreateFinancialTransactionCategoryIT.java

```diff
@@ -0,0 +1,221 @@
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()));
+
+        assertEquals(0, financialTransactionCategoryRepository.count());
+        assertEquals(0, walletRepository.count());
+        assertEquals(1, userRepository.count());
+    }
+
+
+    @DisplayName("Should return error when name is empty")
+    @Test
+    void testCreateNameEmpty() throws Exception {
+        var financialTransactionCategoryCreateDTO = new FinancialTransactionCategoryCreateDTO
+                ("",
+                        FinancialTransactionType.INCOME,
+                        1L);
+
+        mockMvc.perform(post("/api/categories")
+                        .contentType(APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(user("1"))
+                        .with(csrf()))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()));
+
+        assertEquals(0, financialTransactionCategoryRepository.count());
+        assertEquals(0, walletRepository.count());
+        assertEquals(1, userRepository.count());
+    }
+
+    @DisplayName("Should return error when name contains invalid characters")
+    @Test
+    void testCreateNameContainsInvalidCharacters() throws Exception {
+        var financialTransactionCategoryCreateDTO = new FinancialTransactionCategoryCreateDTO
+                ("Catego*&*^ry@",
+                        FinancialTransactionType.INCOME,
+                        1L);
+
+        mockMvc.perform(post("/api/categories")
+                        .contentType(APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(financialTransactionCategoryCreateDTO))
+                        .with(user("1"))
+                        .with(csrf()))
+                .andExpect(status().isBadRequest())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode()));
+
+        assertEquals(0, financialTransactionCategoryRepository.count());
+        assertEquals(0, walletRepository.count());
+        assertEquals(1, userRepository.count());
+    }
+
+    //invalid type
```

Comment:
Invalid types. So below are tests for improper types. And where are the ones for the right ones? I ask because I did not see a related comment above.